src/abi.cpp

```diff
@@ -51,16 +62,23 @@ abi_type* get_type(std::map<std::string, abi_type>& abi_types,
     if (it == abi_types.end()) {
         if (ends_with(name, "?")) {
             auto base = get_type(abi_types, name.substr(0, name.size() - 1), depth + 1);
-            eosio::check(!holds_any_alternative<abi_type::optional, abi_type::array, abi_type::extension>(base->_data),
+            eosio::check(!holds_any_alternative<abi_type::optional, abi_type::array, abi_type::szarray, abi_type::extension>(base->_data),
                   eosio::convert_abi_error(abi_error::invalid_nesting));
             auto [iter, success] = abi_types.try_emplace(name, name, abi_type::optional{base}, &abi_serializer_for< ::abieos::pseudo_optional>);
             return &iter->second;
         } else if (ends_with(name, "[]")) {
             auto element = get_type(abi_types, name.substr(0, name.size() - 2), depth + 1);
-            eosio::check(!holds_any_alternative<abi_type::optional, abi_type::array, abi_type::extension>(element->_data),
+            eosio::check(!holds_any_alternative<abi_type::optional, abi_type::array, abi_type::szarray, abi_type::extension>(element->_data),
                   eosio::convert_abi_error(abi_error::invalid_nesting));
             auto [iter, success] = abi_types.try_emplace(name, name, abi_type::array{element}, &abi_serializer_for< ::abieos::pseudo_array>);
             return &iter->second;
+         } else if(is_szarray(name) ){
+            int pos = name.find_last_of('[');
+            auto element = get_type(abi_types, name.substr(0, pos), depth + 1);
+            eosio::check(!holds_any_alternative< abi_type::szarray, abi_type::extension>(element->_data),
```

Comment:
I believe abi_type::optional is missing here for szarray, like previous parsing on []  which means general array can be optional, sized array say [10] can also be optional

Reply:
Good point, changed.